lib/python/picongpu/pypicongpu/output/openPMD.py

```diff
@@ -0,0 +1,96 @@
+"""
+This file is part of PIConGPU.
+Copyright 2021-2024 PIConGPU contributors
+Authors: Julian Lenz
+License: GPLv3+
+"""
+
+from typing import Optional
+from .. import util
+from ..rendering import RenderedObject
+
+import typeguard
+
+
+@typeguard.typechecked
+class OpenPMD(RenderedObject):
+    """
+    Class to configure openPMD output.
+
+    This class requires a period (in time steps) and will enable OpenPMD output.
+    """
+
+    period = util.build_typesafe_property(int)
+    """period to print data at"""
+
+    source = util.build_typesafe_property(str)
+    """data sources and filters to dump"""
```

Comment:
I am not really happy with the representation of data sources as strings, we should use objects to define the sources, and then generate a matching `fileOutput.param` and cfg block based on them,

... but I would be willing to do that in a later PR.